analysis/11_an_stand_surv_asthma.do

```diff
@@ -0,0 +1,78 @@
+/*==============================================================================
+DO FILE NAME:			11_an_stand_surv_asthma
+PROJECT:				ICS in COVID-19 
+AUTHORS:				A Schultze
+						adapted from K Baskharan A Wong
+DATE: 					12th of May 2020
+VERSION: 				Stata 16.1 
+DESCRIPTION OF FILE:	fit parametric model
+						use postestimation command to plot standardised surv 
+						estimate difference in % at 2 months
+						derive marginal HR
+						
+DATASETS USED:			$tempdir\analysis_dataset_STSET_$outcome.dta
+DATASETS CREATED: 		None
+OTHER OUTPUT: 			Results in svg: $outdir\standsurvplot1
+						Log file: $logdir\11_an_stand_surv_asthma 
+						
+==============================================================================*/
+
+* Open a log file
+capture log close
+log using $logdir\11_an_stand_surv_asthma, replace t
+
+* Open Stata dataset
+use $tempdir\analysis_dataset_STSET_$outcome, clear
+
+/* Sense check outcomes=======================================================*/ 
+tab exposure $outcome
+
+/* Fit the parametric model===================================================*/
+
+xi i.exposure i.male $varlist
+stpm2 _I* age1 age2 age3, scale(hazard) df(3) eform nolog stratify(stp)
+
+/* Predict survival===========================================================*/
+
+* set timevar for time points to be plotted
+summ _t
+local tmax=r(max)
+local tmaxplus1=r(max)+1
+
+display `tmax'
+
+range timevar 0 `tmax' `tmaxplus1'
+
+* Generate the standardised predictions 
+stpm2_standsurv, at1(_Iexposure_1 0 _Iexposure_2 0) at2(_Iexposure_1 1) at3(_Iexposure_2 1) timevar(timevar) ci contrast(difference) fail
```

Comment:
I've been thinking about how to work out for categorial exposure (level >2). I'm not sure about this too, need to check with @krishnanbhaskaran

Reply:
Thanks angel, I updated this code following an email response from Krishnan to at1(_Iexposure_1 0 _Iexposure_2 0) at1(_Iexposure_1 1 _Iexposure_2 0) at1(_Iexposure_1 0 _Iexposure_2 1); but he thinks it should be the equivalent of what's written here.

Reply:
great, good to know!